api/envoy/config/cluster/redis/redis_cluster.proto

```diff
@@ -51,4 +58,17 @@ message RedisClusterConfig {
   // Timeout for topology refresh request. If not set, this defaults to 3s.
   google.protobuf.Duration cluster_refresh_timeout = 2
       [(validate.rules).duration.gt = {}, (gogoproto.stdduration) = true];
+
+  // The minimum interval that must pass after triggering a topology refresh request before a new
+  // request can possibly be triggered again, no matter how many redirection errors are received
+  // from upstream servers during this interval. If not set, this defaults to 5s.
+  google.protobuf.Duration redirect_refresh_interval = 3 [(gogoproto.stdduration) = true];
+
+  // The number of redirection errors received in the last minute that must be reached before
+  // triggering a topology refresh request. If not set, this defaults to 0. In this case every
+  // redirection error will trigger a topology refresh, limited only by when the last error was
+  // received and the
+  // :ref:`redirect_refresh_interval<envoy_api_field_config.cluster.redis.RedisClusterConfig.redirect_refresh_interval>`
+  // setting.
+  uint32 redirect_per_minute_refresh_threshold = 4;
```

Comment:
I think a simple redirect_refresh_threshold is easier to reason about and easier to implement.  So this would be the number of redirects since the last fresh.  If we just do a straight count, the size of cluster and request rate doesn't impact the configuration.  The redirect_refresh_interval would still safe guard the cluster from a thundering herd of refresh.

Reply:
new code up with a simple threshold...